app/Http/Controllers/SupplyController.php

```diff
@@ -225,37 +214,53 @@ public function logNetworkClick(
         return $response;
     }
 
-    public function logNetworkView(Request $request, AdUser $contextProvider, string $bannerId): RedirectResponse
+    private function getRedirectionUrlFromQuery(Request $request): ?string
     {
         if ($request->query->get('r')) {
             $url = Utils::urlSafeBase64Decode($request->query->get('r'));
             $request->query->remove('r');
+
+            return $url;
         }
+    }
```

Comment:
Possibly missing `return null` if 'r' is_empty.
Please, analyze what will happen in logNetworkView while `r` will not be passed. Especially check Utils functions.